src/client/fsharp/FantomasOnline/Decoders.fs

```diff
@@ -33,6 +34,70 @@ let decodeOptions json =
     Decode.object (fun get -> get.Required.Field "settings" (Decode.list optionDecoder))
 
+let rec decodeEditorConfigLine (currentSettings: Map<string, FantomasOption>) i (line: string) =
+    let parts =
+        line.Split([| '='; ' ' |], StringSplitOptions.RemoveEmptyEntries)
+        |> Array.toList
+
+    let toPascalCase (str: string) =
+        let rec buildStr acc (split: string list) =
+            match split with
+            | [] -> acc
+            | h :: t ->
+                let first = h[0] |> Char.ToUpper |> string
+                buildStr (acc + first + h[1..]) t
+
+        str.Split([| '_' |], StringSplitOptions.RemoveEmptyEntries)
+        |> Array.toList
+        |> buildStr ""
+
+    match parts with
+    | name :: value :: _ ->
+        let name = name.Trim().Replace("fsharp_", "")
+        let value = value.Trim()
+        let isInt, intValue = Int32.TryParse value
+        let isBool, boolValue = Boolean.TryParse value
+
+        let pascalName = toPascalCase name
+        let existing = currentSettings.TryFind pascalName
+
+        let optionValue =
+            match existing with
+            | Some(FantomasOption.IntOption(order, pascalName, _)) when isInt ->
+                FantomasOption.IntOption(order, pascalName, intValue)
+            | Some(FantomasOption.BoolOption(order, pascalName, _)) when isBool ->
+                FantomasOption.BoolOption(order, pascalName, boolValue)
+            | Some(FantomasOption.MultilineFormatterTypeOption(order, pascalName, _)) ->
+                FantomasOption.MultilineFormatterTypeOption(order, pascalName, value)
+            | Some(FantomasOption.EndOfLineStyleOption(order, pascalName, _)) ->
+                FantomasOption.EndOfLineStyleOption(order, pascalName, value)
+            | Some(FantomasOption.MultilineBracketStyleOption(order, pascalName, _)) ->
+                FantomasOption.MultilineBracketStyleOption(order, pascalName, value)
+            | _ ->
+                let pascalName = toPascalCase name
+
+                match name with
+                | _ when isInt -> FantomasOption.IntOption(i, pascalName, intValue)
+                | _ when isBool -> FantomasOption.BoolOption(i, pascalName, boolValue)
+                | "record_multiline_formatter"
+                | "array_or_list_multiline_formatter" ->
+                    FantomasOption.MultilineFormatterTypeOption(i, pascalName, value)
+                | "end_of_line" -> FantomasOption.EndOfLineStyleOption(i, pascalName, value)
+                | "multiline_bracket_style" -> FantomasOption.MultilineBracketStyleOption(i, pascalName, value)
+                | name -> failwithf $"Cannot decode %s{name}"
+
+        pascalName, optionValue
+    | _ -> failwithf $"Cannot decode `{line}`"
+
+let decodeOptionsFromEditorConfigFile (currentSettings: Map<string, FantomasOption>) (fileText: string) =
+    fileText.Split([| Environment.NewLine |], StringSplitOptions.RemoveEmptyEntries)
```

Comment:
This won't quite work when there are multiple sections.
This will assume all settings found will apply.
Might be confusing for the users.